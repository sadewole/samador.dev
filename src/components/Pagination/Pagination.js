import React from "react"
import { Link } from "gatsby"

const Pagination = ({
  nextPagePath,
  prevPagePath,
  hasPrevPage,
  hasNextPage,
}) => {
  return (
    <div className="my-10 flex items-center justify-between">
      <h2>
        <Link
          rel="prev"
          to={hasPrevPage ? prevPagePath : "/blogs"}
          className={`uppercase hover:text-white font-semibold hover:no-underline ${
            !hasPrevPage && "pointer-events-none text-white"
          }`}
        >
          ← prev
        </Link>
      </h2>
      <h2>
        <Link
          rel="next"
          to={hasNextPage ? nextPagePath : "/blogs"}
          className={`uppercase hover:text-white font-semibold hover:no-underline ${
            !hasNextPage && "pointer-events-none text-white"
          }`}
        >
          → next
        </Link>
      </h2>
    </div>
  )
}

export default Pagination
