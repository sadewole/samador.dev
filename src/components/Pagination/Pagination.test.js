import React from "react"
import { render } from "@testing-library/react"
import renderer from "react-test-renderer"
import Pagination from "./Pagination"

describe("Pagination", () => {
  const props = {
    nextPagePath: "/test",
    prevPagePath: "/test",
    hasPrevPage: false,
    hasNextPage: false,
  }

  it("must be string", () => {
    const { getByText } = render(<Pagination {...props} />)

    getByText("← prev")
    getByText("→ next")
  })

  it("render correctly", () => {
    const tree = renderer.create(<Pagination {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
