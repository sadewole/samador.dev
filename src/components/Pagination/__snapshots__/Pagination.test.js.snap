// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination render correctly 1`] = `
<div
  className="my-10 flex items-center justify-between"
>
  <h2>
    <a
      className="uppercase hover:text-white font-semibold hover:no-underline pointer-events-none text-white"
      href="/blogs"
      rel="prev"
    >
      ← prev
    </a>
  </h2>
  <h2>
    <a
      className="uppercase hover:text-white font-semibold hover:no-underline pointer-events-none text-white"
      href="/blogs"
      rel="next"
    >
      → next
    </a>
  </h2>
</div>
`;
